### PR TITLE
Update `just` grammar and queries

### PR DESCRIPTION
**Pull request by tgross35**
Bump the git revision and use the queries from <https://github.com/IndianBoy42/tree-sitter-just/tree/80d612d198b8d6207416225c9fcea0fe86ffb96e/queries-flavored/helix> to update Helix's sources.

Tree-sitter-just has queries for Helix in-tree, I will make any changes from this review upstream.

Originally added at https://github.com/helix-editor/helix/pull/6453 (cc @VuiMuich)

Fixes: #8226

### CODE REVIEW

**Comment by woojiq** [comment]
I don't have a lot of Justfiles, but judging by a quick review, I can tell that it is much better than on master. I also appreciate the comments in queries which help to understand them faster.
I once wanted to bump the version of tree-sitter-just because there were no working injection queries. Now something like this works fine: `set shell := ["fish", "-c"]`, but I still wonder if it is possible to support injections inside a recipe:
```
foo:
    #!/usr/bin/env fish
    function lst
        math "0+$argv[-1]" 2&>/dev/null
    end
```

File: runtime/queries/just/locals.scm
Comment (the-mikedavis) [inline runtime/queries/just/locals.scm:1]: For locals queries we follow upstream tree-sitter and only use `@local.scope`, `@local.definition` and `@local.reference`: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

**Comment by the-mikedavis** [comment]
I believe those injections should work but currently the pattern that defaults other `recipe_body`s to bash is taking precedence no matter which order the patterns are defined in. (I think because the shebang pattern is more specific, so it matches last.) It might work to add a field for `shebang` so you could match against the absence of the field:

```scheme
(recipe_body
  !shebang
  (#set! injection.language "bash")
  (#set! injection.include-children)) @injection.content
(recipe_body
  shebang: (shebang) @injection.shebang
  (#set! injection.include-children)) @injection.content
```

**Comment by tgross35** [comment]
> I believe those injections should work but currently the pattern that defaults other `recipe_body`s to bash is taking precedence no matter which order the patterns are defined in. (I think because the shebang pattern is more specific, so it matches last.) It might work to add a field for `shebang` so you could match against the absence of the field:
> 
> ```scheme
> (recipe_body
>   !shebang
>   (#set! injection.language "bash")
>   (#set! injection.include-children)) @injection.content
> (recipe_body
>   shebang: (shebang) @injection.shebang
>   (#set! injection.include-children)) @injection.content
> ```

Thanks, I did mean to ask about this specifically since it seemed like order in the query file did not matter. I'll play with this a bit - is the `!` syntax standard? I don't see it documented on the tree-sitter website.

**Comment by tgross35** [comment]
Another question, is there any good way to test indents, injections, and textobjects queries?

**Comment by the-mikedavis** [comment]
It's standard, see "negated fields" here under the docs on the query format: https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax. It's a small section that's easy to miss.

Currently we don't have unit tests for queries. We could add something like the query testing in tree-sitter-cli to the xtask crate but it would add a lot of code for something that is usually not too much work to verify by hand

**Comment by tgross35** [comment]
Using `!shebang` turned out pretty great based on tests from https://github.com/IndianBoy42/tree-sitter-just/pull/150. Default to bash:

image

Specifying via shebang:

image

Setting globally:

image

But Helix doesn't seem to pick up the shebang `#!/bin/sh` for some reason. I don't know why this would be since it is covered in https://github.com/helix-editor/helix/blame/6c4d986c1b1ac4e350dced513b6608ba4464cde3/languages.toml#L911 and picked up by the regex in https://github.com/helix-editor/helix/blob/6c4d986c1b1ac4e350dced513b6608ba4464cde3/helix-core/src/syntax.rs#L2509

image

It doesn't pick up `#!/bin/dash` either, but does pick up `#!/bin/bash`.

**Comment by RoloEdits** [comment]
Has this been confirmed to fix #8226 ?

**Comment by tgross35** [comment]
> Has this been confirmed to fix #8226 ?

I believe so from a quick test

**Comment by the-mikedavis** [comment]
Could you push the latest changes to this branch so I could debug? I see the `#!/bin/sh` shebang working on regular files (for example starting `hx file` on a file that only contains the shebang) so I don't see why it wouldn't work in an injection

**Comment by tgross35** [comment]
> Could you push the latest changes to this branch so I could debug? I see the `#!/bin/sh` shebang working on regular files (for example starting `hx file` on a file that only contains the shebang) so I don't see why it wouldn't work in an injection

Sorry @the-mikedavis , this slipped my mind. They are updated now

**Comment by the-mikedavis** [comment]
Looks like the issue is that we look up languages that are tagged with `@injection.shebang` by looking only at language IDs (names): https://github.com/helix-editor/helix/blob/50c90cb47c9cdbb044d1a2de034285e0d198f43e/helix-core/src/syntax.rs#L1024-L1026

We should look up in `language_config_ids_by_shebang` instead like here: https://github.com/helix-editor/helix/blob/50c90cb47c9cdbb044d1a2de034285e0d198f43e/helix-core/src/syntax.rs#L978-L980

**Comment by RoloEdits** [comment]
@tgross35 Where does this pr's status stand? Are the queries all done, just needing the `language_config_ids_by_shebang` fix?

**Comment by tgross35** [comment]
Ah I'm sorry, I just never got back to this. Yeah, to my knowledge it only needs that fix. I haven't looked into it so please feel free to pick that part up (or even this whole PR if you would prefer).

**Comment by RoloEdits** [comment]
with #11306 merged this can now be closed

**Comment by tgross35** [comment]
Oh awesome, thank you for getting this over the line!

**Pull request closed without merging**